src/main/java/org/aksw/simba/lemming/creation/Inferer.java

```diff
@@ -198,15 +217,15 @@ public void iterateStmts(Model newModel, Model sourceModel, OntModel ontModel, M
 			List<? extends OntResource> domain = property.listDomain().toList();
 			for (OntResource curResource : domain) {
 				Statement subjType = ResourceFactory.createStatement(subject, RDF.type, curResource);
-				if (!newModel.contains(subjType)) {
+				if (!model.contains(subjType)) {

```

Comment:
But <code>subjType</code> is also a statement, maybe we need another name without confusion.

Reply:
But subjType is also a statement, maybe we need another name without confusion, like subjNewStmt, objNewStmt

Reply:
What I meant to say is that since Jena's model is a Set of statements, so duplicate statements have no effect on it. Thus, you don't need to check whether the model contains the statement already or not. 

> An RDF Model is represented as a set of statements. (Because a Model is set of statements, adding a duplicate of a statement has no effect.) 
From [Jena Model Tutorial](https://jena.apache.org/tutorials/rdf_api.html#ch-Statements)

Reply:
Ah, got it